source/whats-new.txt

```diff
@@ -22,6 +22,293 @@ What's New in 4.0
+  ideas on our `JIRA project here <https://jira.mongodb.org/projects/NODE>`__.
+
+Key Changes
+~~~~~~~~~~~
+
+Node.js Version
++++++++++++++++
+
+The minimum supported version of Node.js is now v12.9 or greater for
+version 4 of the driver. Support for our 3.x branches will continue
+until summer 2022 to allow time to upgrade.
+
+Cursor Improvements
++++++++++++++++++++
+
+Affected classes:
+
+- ``AbstactCursor``
+- ``FindCursor``
+- ``AggregationCursor``
+- ``ChangeStreamCursor`` (This is the underlying cursor for ``ChangeStream``)
+- ``ListCollectionsCursor``
+
+Our Cursor implentation has been updated to make it clear what is possible
+before and after execution of an operation.
+
+.. example::
+
+   .. code-block:: js
+      :copyable: False
+
+      const fc = collection.find({a: 2.3}).skip(1)
+      for await (const doc of fc) {
+        console.log(doc)
+        fc.limit(1) // incorrect usage, cursor already executing
+      }
+
+There was inconsistency surrounding how the cursor would error if a
+setting was applied after cursor execution began. Now, the cursor will
+throw an error when attempting to apply operations in an invalid state,
+similar to the following:
+
+``MongoError: Cursor is already initialized``
+
+Cursor Stream API
+^^^^^^^^^^^^^^^^^
+
+Our Cursor types no longer extend ``Readable`` directly. They must be
+transformed into a stream by calling ``cursor.stream()``.
+
+.. example::
+
+   .. code-block:: js
+      :copyable: False
+
+      const cursor = collection.find({})
+      const stream = cursor.stream()
+      stream.on("data", data => console.log)
+      stream.on("error", () => client.close())
+
+Use ``hasNext()`` and ``next()`` for manual iteration.
+Use ``for await of`` syntax or any ``Promise`` helpers for
+asynchronous iteration.
+
+``MongoClientOptions`` Interface
++++++++++++++++++++++++++++++++++
+
+With type hinting, you should find that options passed to a ``MongoClient``
+are enumerated and discoverable. We've made a large effort to process
+all options in the driver to give early warnings about incompatible settings
+to get your app up and running in a correct state quickly.
+
+- ``checkServerIdentity`` is no longer checked before being passed to the
+  underlying Node API. Previously, accepted values were ``false``, or
+  a function. Now, the argument must be a function. Specifying a
+  boolean will result in an error being thrown.
+
+``createCollection()``
+++++++++++++++++++++++
+
+This method no longer supports a ``strict`` option, which returned
+an error if the collection did not exist. To assert the existence of
+a collection, use the ``listCollections()`` method instead.
+
+.. example::
+
+   .. code-block:: js
+
+      const collections = (await db.listCollections({}, { nameOnly: true })
+        .toArray()).map(
+          ({name}) => name
+        );
+      if (!collections.includes(myNewCollectionName)) {
+        await db.createCollection(myNewCollectionName);
+      }
```

Comment:
Recopy this snippet, in order to replicate the old behavior we should actually throw inside the if stmt.